dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerEMQHelper.java

```diff
@@ -0,0 +1,38 @@
+package com.ibm.ws.collector.manager.buffer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import com.ibm.wsspi.collector.manager.BufferManager;
+
+public class BufferManagerEMQHelper {
+
+    protected volatile static boolean EMQRemovedFlag = false;
+    private static final int EMQ_TIMER = 60 * 5 * 1000; //5 Minute timer
+    protected static List<BufferManager> bufferManagerList= Collections.synchronizedList(new ArrayList<BufferManager>());
+
+    public BufferManagerEMQHelper() {
+    }
+
+    public static void removeEMQByTimer(){
+        new java.util.Timer().schedule(
+                new java.util.TimerTask() {
+                    @Override
+                    public void run() {
+                        BufferManagerEMQHelper.removeEMQTrigger();
+                    }
+                },
+                BufferManagerEMQHelper.EMQ_TIMER);
+    }
+    
+    public synchronized static void removeEMQTrigger(){
+            BufferManagerEMQHelper.EMQRemovedFlag=true;
+            for(BufferManager i: BufferManagerEMQHelper.bufferManagerList) {
```

Comment:
You need to have `synchronize(bufferManagerList)` for this to make this thread safe

Reply:
Created an add method in BufferManagerEMQHelper which is synchronized and adds to the bufferManagerList.